component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/AbstractMetricEventBuilder.java

```diff
@@ -40,5 +40,5 @@ public Map<String, Object> build() throws MetricReportingException {
      *
      * @return Map representing attributes of Metric Event
      */
-    protected abstract Map<String, Object> buildEvent();
+    protected abstract Map<String, Object> buildEvent() throws MetricReportingException;
```

Comment:
What is the purpose of this reporting exception. When are we throwing that? By design build event is not supposed to throw checked exceptions

Reply:
Currently it is throwing from the validation level.
By validation I mean the validation happening inside the building phase..
I will handle it in that level.

Reply:
building phase should not do validation. Validation should be done at the validate level

Reply:
Ack.